js/lex.go

```diff
@@ -66,21 +90,34 @@ func (tt TokenType) String() string {
 
 // Lexer is the state for the lexer.
 type Lexer struct {
-	r *buffer.Lexer
-
-	regexpState   bool
-	templateState bool
+	r     *buffer.Lexer
+	stack []ParsingContext
```

Comment:
I meant instead of `stack` as that doesn't convey the meaning of the variable, only the way it is operated